### ISSUE TITLE
Training Challenges

### ISSUE THREAD

**Issue by theATM** [ZhuangShuoH/geospatial-object-detection#3]
Hi, I have been trying to reproduce results from this paper. Sadly, I stabled apron some challenges. 
During my trainings (max 20 epochs) the model accomplished minimal loss 7,5. Sadly when running evaluation on said model all classes returned average precision equaled 0.0 for all classes. 
What is more strange i run some trainings with racoon and kangaroo datasets and they were giving promising results ( average precision ~0.75), though these are single class datasets.
Maybe I am using the wrong backend model. That is why I must ask which pretrained weights for backend have been used during training and on which dataset were they trained on. Are those [these](https://bit.ly/39rLNoE) - backend.h5? 
Thanks in advance.

**Comment by Dr-Zhuang** [comment]
Yes, the backend.h5 was used as a pre-trained model. According to your description, the minimal training loss was about 7.5. It seems that there was no problem during the training process. You can use the well-trained model (can be found https://github.com/Dr-Zhuang/geospatial-object-detection/issues/2) to evaluate the RSD-GOD dataset. If the returned APs are still false, you should check the evaluation.py and  config.json.

**Comment by theATM** [comment]
Thanks for the answer. 
The problem I faced was entirely on me. I tried to make this code run on modern GPU which simply did not supported the Tensorflow 1.x, but rather than print an error message it simply performed incorrect calculations resulting, sometimes (but not always), in the nan loss. Zero score in the evaluation may have been related to this issue.
(or simply there were no annotations in the cache file due to my grammar bugs etc)
To fix this issue I "simply" used docker (based on the nvcr.io/nvidia/tensorflow:22.11-tf1-py3) and it worked, but cuda training was somewhat slow.
 
Overall I managed to recreate the training on the cuda, which after running for the 57/100 epochs and stopping ( due to early stopping, loss = 8.93038) and the results are:

On Validation:
airport: 0.8388
helicopter: 0.9376
oiltank: 0.9217
plane: 0.8605
warship: 0.7732
mAP@50: 0.8664

Of course my model was worse then the pretrained one, but I wasn't aiming for the exact copy and I did not sufficiently tune the hyperparameters. I just wanted to make the pipeline work.

After all, thanks for the help and congratulation on the great work.
